cc/BUILD.bazel

```diff
@@ -0,0 +1,20 @@
+load("@modular_cc_toolchain//cc:features.bzl", "cc_feature", "action_mux")
+
+load("@rules_cc//cc:action_names.bzl", "ALL_CC_COMPILE_ACTION_NAMES", "ALL_CC_LINK_ACTION_NAMES")
+
+cc_feature(
+    name = "garbage_collect_sections",
+    action_flags = action_mux({
+        tuple(ALL_CC_COMPILE_ACTION_NAMES): ["-ffunction-sections", "-fdata-sections"],
```

Comment:
There were a couple of quirks here, that I've just hacked around for now. Namely, the action group lists aren't hashable so I've just converted them to tuples for the moment. But the plan is to get this looking a little nicer.